include/sch/S_Object/S_Cone.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+#ifndef _S_CONE_H_SCH
+#define _S_CONE_H_SCH
+
+#include <sch/S_Object/S_ObjectNonNormalized.h>
+
+namespace sch
+{
+
+  class S_Cone :
+    public S_ObjectNonNormalized
+  {
+  public:
+    /*!
+    * \brief constructor
+    *  the cone centered at the apex and having its axis aligned on the z axis
+    * and directed above
+    * \param angle is the angle between the axis of the cone and the side of
+    *         the cone. Make sure the angle is less than pi/2 
+    *         otherwise the behavior is not defined
+    * \param height is the height of the cone.
+    */
+    SCH_API S_Cone(const Scalar& angle, const Scalar& height );
+    SCH_API virtual ~S_Cone();
+
+    SCH_API virtual Point3 l_Support(const Vector3& v, int& lastFeature)const;
+
+    SCH_API virtual S_ObjectType getType() const;
+
+    SCH_API virtual Scalar getAngle() const;
+    SCH_API virtual Scalar getHeight() const;    
+
+  protected:
+    Scalar angle_;
+    Scalar height_;
+    Scalar angleTan_;
+    Scalar baseRadius_;
+
+    static const Point3 origin_;
```

Comment:
I don't think it's useful to have a static member for this, since `l_Support` returns a value anyway this brings no benefit

Reply:
Indeed, but I wanted to use it to make a displaylist on sch-visualization. I guess I won't have time for that right now. I remove it and will put it back then.

Reply:
Solved with https://github.com/mehdi-benallegue/sch-core/commit/c91a14a99e32baff748f9468e2490ad40d8fa867